Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1492,12 +1526,31 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes()
         }
     };
 
-    auto computeIndefiniteItems = [&](SingleThreadWeakListHashSet<RenderBox>& indefiniteItems) {
+    auto computeIndefiniteItems = [&](MasonryIndefiniteItems &masonryIndefiniteItems) {
         for (auto trackIndex : m_contentSizedTracksIndex) {
             GridIterator iterator(m_grid, m_direction, trackIndex);
             while (CheckedPtr<RenderBox> gridItem = iterator.nextGridItem()) {
-                if (GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, *gridItem, m_direction).isIndefinite())
-                    indefiniteItems.add(*gridItem);
+                if (GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, *gridItem, m_direction).isIndefinite()) {
+                    auto spanLength = m_renderGrid->gridSpanForChild(*gridItem, m_direction).integerSpan();
+
+                    bool isSubgrid = false;
+                    if (gridItem->isRenderGrid() && downcast<RenderGrid>(gridItem.get())->isSubgridInParentDirection(iterator.direction()))
+                        isSubgrid = true;
```

Comment:
You might be able to refactor this into a single line if you think it doesn't harm readability